sdk/storage/src/datalake/path_client.cpp

```diff
@@ -0,0 +1,385 @@
+        m_pathUri.to_string(), *m_pipeline, options.Context, protocolLayerOptions);
+  }
+
+  PathCreateResponse PathClient::Create(const PathCreateOptions& options) const
+  {
+    DataLakeRestClient::Path::CreateOptions protocolLayerOptions;
+    // TODO: Add null check here when Nullable<T> is supported
+    protocolLayerOptions.Resource = options.Resource;
+    protocolLayerOptions.Continuation = options.Continuation;
+    protocolLayerOptions.Mode = options.Mode;
+    protocolLayerOptions.SourceLeaseId = options.SourceLeaseId;
+    protocolLayerOptions.LeaseIdOptional = options.LeaseId;
+    protocolLayerOptions.CacheControl = options.CacheControl;
+    protocolLayerOptions.ContentType = options.ContentType;
+    protocolLayerOptions.ContentDisposition = options.ContentDisposition;
+    protocolLayerOptions.ContentEncoding = options.ContentEncoding;
+    protocolLayerOptions.ContentLanguage = options.ContentLanguage;
+    protocolLayerOptions.IfMatch = options.IfMatch;
+    protocolLayerOptions.IfNoneMatch = options.IfNoneMatch;
+    protocolLayerOptions.IfModifiedSince = options.IfModifiedSince;
+    protocolLayerOptions.IfUnmodifiedSince = options.IfUnmodifiedSince;
+    protocolLayerOptions.SourceIfMatch = options.SourceIfMatch;
+    protocolLayerOptions.SourceIfNoneMatch = options.SourceIfNoneMatch;
+    protocolLayerOptions.SourceIfModifiedSince = options.SourceIfModifiedSince;
+    protocolLayerOptions.SourceIfUnmodifiedSince = options.SourceIfUnmodifiedSince;
+    protocolLayerOptions.Properties = SerializeMetadataString(options.Metadata);
+    protocolLayerOptions.Umask = options.Umask;
+    protocolLayerOptions.RenameSource = options.RenameSource;
+    protocolLayerOptions.Permissions = options.Permissions;
+    protocolLayerOptions.Timeout = options.Timeout;
+    return DataLakeRestClient::Path::Create(
+        m_pathUri.to_string(), *m_pipeline, options.Context, protocolLayerOptions);
+  }
+
+  PathDeleteResponse PathClient::Delete(const PathDeleteOptions& options) const
+  {
+    DataLakeRestClient::Path::DeleteOptions protocolLayerOptions;
+    // TODO: Add null check here when Nullable<T> is supported
+    protocolLayerOptions.Continuation = options.Continuation;
+    protocolLayerOptions.LeaseIdOptional = options.LeaseId;
+    protocolLayerOptions.IfMatch = options.IfMatch;
+    protocolLayerOptions.IfNoneMatch = options.IfNoneMatch;
+    protocolLayerOptions.IfModifiedSince = options.IfModifiedSince;
+    protocolLayerOptions.IfUnmodifiedSince = options.IfUnmodifiedSince;
+    protocolLayerOptions.RecursiveOptional = options.RecursiveOptional;
+    protocolLayerOptions.Timeout = options.Timeout;
+    return DataLakeRestClient::Path::Delete(
+        m_pathUri.to_string(), *m_pipeline, options.Context, protocolLayerOptions);
+  }
+
+  GetPathPropertiesResponse PathClient::GetProperties(const PathGetPropertiesOptions& options) const
+  {
+    DataLakeRestClient::Path::GetPropertiesOptions protocolLayerOptions;
```

Comment:
just out of curiosity, why not aggregate initialize `protocolLayerOptions`?

I don't mean i think we should do this, but we should be consistent, right?

Reply:
Think of the original plan of using options, which is to avoid breaking change for the convenience layer to have to adapt to protocol layer change, with possible shifting sequence of the optional parameters. If we aggregate initialize anything from the protocol layer in the convenience layer, we have to upgrade the convenience layer code to adapt to possible protocol layer change as well, this is what I wanted to avoid.

Reply:
okay